### PR TITLE
[SPARK-14819] [SQL] Improve SET / SET -v command

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

Currently `SET` and `SET -v` commands are similar to Hive `SET` command except the following difference:
1. The result is not sorted;
2. When using `SET` and `SET -v`, in addition to the Hive related properties, it will also list all the system properties and environment properties, which is very useful in some cases.

This JIRA is trying to make the current `SET` command more consistent to Hive output.
## How was this patch tested?

Current test cases SQLQuerySuite and HiveQuerySuite should have enough test coverage.

### CODE REVIEW

**Comment by rxin** [comment]
You might need to rebase this in a bit because i'm refactoring all of these quite heavily for Spark 2.0.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 1]
```diff
@@ -716,4 +716,8 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
     }
   }
 
+  test("set / set -v") {
+    checkExistence(sql("set"), true, "env:", "system:")
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: can we add a test case that checks the value of SPARK_TESTING?

Comment (bomeng) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: SPARK_TESTING is not in sys.env nor sys.props while running the test cases. Any suggestions to add the test cases for that?

Comment (bomeng) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: Found out that current test cases SQLQuerySuite and HiveQuerySuite have intensive test coverage for SET command.

**Comment by rxin** [comment]
@bomeng while you are at this, do you mind implementing https://issues.apache.org/jira/browse/SPARK-14806 together?

**Comment by dongjoon-hyun** [comment]
Hi, @bomeng . 
I hit the same issue. Could you update the PR?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala [hunk 2]
```diff
@@ -138,17 +138,18 @@ case class SetCommand(kv: Option[(String, Option[String])]) extends RunnableComm
...
     case None =>
       val runFunc = (sqlContext: SQLContext) => {
-        sqlContext.getAllConfs.map { case (k, v) => Row(k, v) }.toSeq
+        sqlContext.getAllConfs.toSeq.sortBy(_._1).map { case (k, v) => Row(k, v) } ++
+          getEnvList(withDoc = false)
       }
       (keyValueOutput, runFunc)
 
     // Queries all properties along with their default values and docs that are defined in the
     // SQLConf of the sqlContext.
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala]: We should update the comment regarding the system environment properties.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala [hunk 3]
```diff
@@ -138,17 +138,18 @@ case class SetCommand(kv: Option[(String, Option[String])]) extends RunnableComm
     // Queries all key-value pairs that are set in the SQLConf of the sqlContext.
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala]: Same.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala [hunk 4]
```diff
@@ -182,4 +183,18 @@ case class SetCommand(kv: Option[(String, Option[String])]) extends RunnableComm
 
   override def run(sqlContext: SQLContext): Seq[Row] = runFunc(sqlContext)
 
+  /**
+   * get the system environment properties as a sequence
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala]: Nit: "get" -> "Gets", "as a sequence" -> "as a sequence of rows".

**Comment by dongjoon-hyun** [comment]
Hmm. This PR seems to be stale. Actually, what I wanted was a **sorted** result of `SET` and `SET -v`.

**Comment by dongjoon-hyun** [comment]
I made another #16579  for `SET` for a sorted result.

**Comment by HyukjinKwon** [comment]
@dongjoon-hyun, then, do you mind if I ask if this PR is not useful in favour of #16579?

**Comment by dongjoon-hyun** [comment]
Hi, @bomeng , do you have something to refresh here?
I think we can close this for now.

**Pull request closed without merging**